### DISCUSSION TITLE
Email Confirm and Password reset via code

### DISCUSSION THREAD

**Discussion by nomanmurtaza786** [supabase/supabase#904] [Questions]
Hi Team, 
Is it possible for email confirmation we ask users to enter the verification code rather than clicking the link?

**Comment by silentworks** [comment]
I don't think this is possible as a feature of supabase auth right now, but you could implement this yourself by disabling the email confirmation setting in your supabase admin UI and then creating your own flags to control this. I wouldn't recommend going this route though.

**Comment by GaryAustin1** [comment]
I can confirm per the latest blog post: https://supabase.com/blog/2022/03/28/community-day (scroll to Send OTP via email)

Email OTP is working!

Email on signup with template modified with {{ .Token}}
![image](https://user-images.githubusercontent.com/54564956/160448771-a1f0b508-3688-4c32-a24a-8cc992e5db47.png)

And call made to confirm the user.
```
const {user,session,error} = await supabase.auth.verifyOTP ({
	email:'name@email.com',
	token:'owOWaODcyzG0ItuHQ24ipA',
	type:'signup'
})
```

**Reply by EthanTrevino** [reply]
That's pretty close. Is there a way to maybe send just the last 6 characters via email? Having a user enter that whole string is a lot to ask.

**Reply by GaryAustin1** [reply]
This seems to be an effort to shorten it to 10 characters.  
https://github.com/supabase/gotrue/pull/446

**Comment by EthanTrevino** [comment]
Thank you so much. I'll keep an eye on that PR.
